chaoslib/openebs/cstor_pool_failure.yaml

```diff
@@ -0,0 +1,73 @@
+- name: Derive PV from application PVC 
+  shell: >
+    kubectl get pvc {{ app_pvc }}
+    -o custom-columns=:spec.volumeName -n {{ app_ns }}
+    --no-headers
+  args:
+    executable: /bin/bash
+  register: pv
+
+- name: Derive SC from application PVC 
+  shell: >
+    kubectl get pvc {{ app_pvc }}
+    -o custom-columns=:spec.storageClassName -n {{ app_ns }}
+    --no-headers
+  args:
+    executable: /bin/bash
+  register: sc
+
+- name: Derive ReplicaCount from SC
+  shell: >
+    kubectl get sc {{ sc.stdout }} -n {{ operator_ns }} -o json |
+    jq '.metadata.annotations."cas.openebs.io/config"' -r |
+    grep -A 1 -w "name: ReplicaCount" | grep -w value | awk '{print $2}'
+  args:
+    executable: /bin/bash
+  register: replicacount
```

Comment:
Shall we derive this using deployment.

Reply:
deployment will not have replicacount

Reply:
The grep becomes a necessity here as we hit a dead-end upon extracting the content (params) from the cas.openebs.io/config annotation !